### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by maiqueb**
Fixes: #9 

Here's the [doc preview](https://github.com/maiqueb/multus-dynamic-networks-controller/blob/483a05e4d5807e6c1c77eff5fffc2e348b98c3b5/README.md).

This PR updates the project's README, indicating:
- project's description
- how to install
- how to remove
- how to use
- developer info (getting your code changes in a live cluster)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,142 @@
...
+      command: ["/bin/sleep", "10000"]
+```
+
+## Configuration
+The `multus-dynamic-networks-controller` configuration is encoded in JSON, and allows the following keys:
+
+- `"criSocketPath"`: specify the path to the CRI socket. Defaults to `/run/containerd/containerd.sock`.
+- `"criType"`: either `crio` or `containerd`. Defaults to `containerd`.
+- `"multusSocketDir"`: directory where the multus socket will be found. Defaults to `/var/run/multus-cni/`.
+
```
Comment (orelmisan) [inline README.md:133]: Please consider adding a paragraph explaining how to view the controller's logs.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,142 @@
...
+## Developer workflow
+Below you can find information on how to push local code changes to a kind cluster.
+
+- change code :)
+- start up a kind cluster. I've tested using the [multus repo e2e kind cluster](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/e2e/setup_cluster.sh).
```
Comment (orelmisan) [inline README.md:138]: Is this kind cluster different from a vanilla one?

File: examples/interface-add.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
```
Comment (orelmisan) [inline examples/interface-add.yaml:1]: I suggest to drop the example manifests if you also have them in the README file.
This way you don't have to maintain two copies of the same thing.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,142 @@
...
+```
+
+## Configuration
+The `multus-dynamic-networks-controller` configuration is encoded in JSON, and allows the following keys:
+
+- `"criSocketPath"`: specify the path to the CRI socket. Defaults to `/run/containerd/containerd.sock`.
+- `"criType"`: either `crio` or `containerd`. Defaults to `containerd`.
+- `"multusSocketDir"`: directory where the multus socket will be found. Defaults to `/var/run/multus-cni/`.
+
+## Developer workflow
```
Comment (orelmisan) [inline README.md:134]: I suggest to capitalize the first character in each step's first word.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,142 @@
...
+
+### Installation
+Use the provided [manifest](manifests/dynamic-networks-controller.yaml) to install the controller on your cluster.
+
+```bash
+kubectl apply -f manifests/dynamic-networks-controller.yaml
+```
+
+Currently, it doesn't do much - it doesn't invoke the delegates.
+As such, it'll only watch for multus network annotations changes.
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply README.md]: Done.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,142 @@
 # multus-dynamic-networks-controller
-A controller listening to pod's network selection elements. When the multus annotations changes, it will invoke the corresponding delegate
+This project provides a Kubernetes controller granting the ability to plug/unplug network interfaces
+to / from running pods.
+
+This controller is listening to pod's network selection elements, and whenever those change
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply README.md]: Done.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,142 @@
...
+    app: macvlan
+spec:
+  containers:
+    - name: macvlan-worker1
+      image: docker.io/library/alpine:latest
+      command: ["/bin/sleep", "10000"]
+```
+
+## Configuration
+The `multus-dynamic-networks-controller` configuration is encoded in JSON, and allows the following keys:
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply README.md:134]: Done.

I'm afraid I don't know how to link to a line to a file in the repo (until the file is committed to the remote ...).

If you happen to know how, I'll link it properly.

File: README.md (hunk 1 above)
Comment (maiqueb) [reply README.md:144]: It's just a regular pod ... meaning you access logs via `kubectl logs -n<namespace> <pod name>` . 

Do you think that info is relevant ? It seems basic.

**Review by maiqueb — COMMENTED** [review]
Thanks for the review @orelmisan .

Good suggestions.

File: examples/interface-add.yaml (hunk 3 above)
Comment (orelmisan) [reply examples/interface-add.yaml:1]: I find it easy to use manifests directly from the text using the shell's [here doc](https://en.wikipedia.org/wiki/Here_document#:~:text=In%20computing%2C%20a%20here%20document,it%20were%20a%20separate%20file.) feature:
```bash
cat <<EOF | kubectl apply -f -
---
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: macvlan1-config
spec:
  config: '{
            "cniVersion": "0.4.0",
            "plugins": [
                {
                    "type": "macvlan",
                    "capabilities": { "ips": true },
                    "master": "eth1",
                    "mode": "bridge",
                    "ipam": {
                        "type": "static"
                    }
                }, {
                    "type": "tuning"
                } ]
        }'
EOF
```

File: README.md (hunk 1 above)
Comment (orelmisan) [reply README.md:144]: I agree that it is basic.

> meaning you access logs via kubectl logs -n<namespace> <pod name>

Yeah, but what are they?
A user will have to look at the deployment manifest to get them.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,142 @@
 # multus-dynamic-networks-controller
-A controller listening to pod's network selection elements. When the multus annotations changes, it will invoke the corresponding delegate
+This project provides a Kubernetes controller granting the ability to plug/unplug network interfaces
+to / from running pods.
+
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply README.md:4]: Yes. Thank you.

File: examples/interface-add.yaml (hunk 3 above)
Comment (maiqueb) [reply examples/interface-add.yaml:1]: Yes, but how will any automation consume that from the readme ?

File: README.md (hunk 2 above)
Comment (maiqueb) [reply README.md:149]: It is not actually required - you can use whatever you want. 

... but everything will be connected in the underlying network. Without multiple interfaces in the host, that's not happening.

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (orelmisan) [reply README.md:134]: If it works on your computer, I think it will work properly when the PR is merged.

File: README.md (hunk 1 above)
Comment (maiqueb) [reply README.md:144]: Can you rephrase ? I don't follow. 

Who's "they" ? The logs ?

Comment (orelmisan) [reply README.md:144]: The names of the controller's pod, and the name of the namespace it is deployed to.

Comment (maiqueb) [reply README.md:144]: It will obviously depend of which namespace you deploy the controller to, and what the ds name is. 

I still feel that info is pretty obvious to a user.

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (maiqueb) [reply README.md:134]: If I don't know what to do, I can't write it :) 

Is that even doable ? Link to a line on repo's file ? I can link to anchors in markdown, but I honestly don't know how to do that with file lines.

File: README.md (hunk 1 above)
Comment (orelmisan) [reply README.md:144]: Fair enough.

**Review by orelmisan — APPROVED** [review]
Thank you for the changes!

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (orelmisan) [reply README.md:134]: Haven't seen the `to a line to a file` part.
It is possible to link to a link to a markdown title in another file.

[the remark this answers is not in the document]
Comment (orelmisan) [reply README.md:134]: I was under the impression that you want to link to a file.

File: examples/interface-add.yaml (hunk 3 above)
Comment (maiqueb) [reply examples/interface-add.yaml:1]: maybe doing but using the example link would be a compromise ...

That would not throw the yaml at the readme users face, but the reader will know what to look for.

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (maiqueb) [reply README.md:134]: Oh I see. I wanted to link to the config map definition within the deployment manifest...

Thank you.
